sdk/state/update.go

```diff
@@ -16,8 +17,10 @@ import (
 
 // CloseAgreementDetails contains the details that the participants agree on.
 type CloseAgreementDetails struct {
-	IterationNumber int64
-	Balance         Amount
+	IterationNumber            int64
+	Balance                    Amount
+	ObservationPeriodTime      time.Duration
```

Comment:
❓ it feels weird to me that we have the observation periods on both the `CloseAgreementDetails` and the `Channel`. Eg. this creates the chance that `channel.latestAuthorizedCloseAgreement.observationPeriodTime` is different than `channel.observationPeriodTime`. Or when trying to find the current channel `O` period do you look at the channel or the latest close agreement?

when proposing and confirming a coordinated close both parties should know to use zero for the observation periods, and instead of needing to validate the observation periods match the validation is done when verifying the signatures.

what if we made the `CloseTxs` func:
`func (c *Channel) CloseTxs(d CloseAgreementDetails, observationPeriod time.Duration, observationPeriodLedgerGap int64)` ? Or something similar

Reply:
> what if we made the `CloseTxs` func:
> `func (c *Channel) CloseTxs(d CloseAgreementDetails, observationPeriod time.Duration, observationPeriodLedgerGap int64)`

I don't think we should have the observation periods on CloseTxs as separate parameters because the values are part of the agreement and so the calling code should expect all the data it needs for the agreement to either be stored inside the Channel or on the agreement object. It'd be more odd to expect calling code to pass in the observation period.

> it feels weird to me that we have the observation periods on both the `CloseAgreementDetails` and the `Channel`

We could remove the observation period from the `Channel` so it only lives inside the agreements, in the same way that the balances live only inside agreements. I think that'd work. I just made that change in bc1c96c. Wdyt?

Reply:
>We could remove the observation period from the Channel so it only lives inside the agreements, in the same way that the balances live only inside agreements. I think that'd work. I just made that change in bc1c96c. Wdyt?

I like it a lot 🙌

Reply:
Full disclosure, part of me worries that this introduces some complexity that is in some ways unnecessary, even though it is useful, and so there's a trade-off here. But let's see how it goes, because having it in the open is better than having it in the Channel. Thanks for leaving this comment 😄.